app/models/cat.rb

```diff
@@ -1,4 +1,14 @@
 class Cat < ActiveRecord::Base
+  include ActiveModel::Validations
+
+  validates :name, length: { minimum: 2, maximum: 32 }
+  validates_uniqueness_of :name
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.